### DISCUSSION TITLE
Simplify update process when update script requires more resources (CPU/RAM)

### DISCUSSION THREAD

**Discussion by remz1337** [community-scripts/ProxmoxVE#83] [Ideas]
There are a few scripts that build the app from source, usually requiring more CPU/RAM to build than to run the app. We should think of a way to run the updates from the host so that resources could be adjusted by the script (without requiring the user to do it manually).

Related to PR https://github.com/community-scripts/ProxmoxVE/pull/79

My initial suggestion would be to create script similar to `update-lxcs.sh`, that would execute the `/usr/bin/update` command in each selected LXC.

The tricky part is knowing when it is needed to increase resource and how much should it be increased. For this, I think we can leverage our brand new JSON database by adding this info

1. pull the json db from github (curl/wget)
2. find the app in the db (could try with hostname or lxc description/tags)
3. check if build resources are specified, and if so, extract them (CPU/RAM)
4. (OPT) update resources and reboot
5. run `pct exec <ID> "/usr/bin/update"`

We could add `build_cpu` and `build_ram` to the JSON, something like this (sorry my JSON skills aren't that great):

```
[
  {
    "catagoryName": "Proxmox VE Tools",
    "collectionId": "n7t0uidut85k9vh",
    "collectionName": "categories",
    "created": "2024-05-03 20:53:12.882Z",
    "expand": {
      "items": [
        {
          "alerts": ["a41orwxt5qz082k"],
          "alpine_script": "",
          "collectionId": "lcr86sab2cxeua1",
          "collectionName": "proxmox_scripts",
          "created": "2024-04-28 22:37:14.653Z",
          "default_cpu": "1",
          "default_hdd": "10",
          "default_login": "",
          "default_ram": "512",
          "build_cpu": "4",
          "build_ram": "4096",
          "port": 0,
           ......
        },
      ]
    }
  }
]
```

**Comment by havardthom** [comment]
I think we should avoid host scripts as much as possible. imo it would be better to just remove the downscaling of resources from scripts and always have the resources that are required for building/installing. It is just a max limit and LXCs only use what they need.

**Comment by MickLesk** [comment]
There are 2 variants:

1. remove the downscaling of resources. However, observe the documentation and point out to users that the scripts may have to be adapted again in the worst case. 

2. add a flag to Advanced Settings (build.func) so that they are not automatically scaled down. (original error).
